### DISCUSSION TITLE
Editor toolbar button to toggle drawing canvas

### DISCUSSION THREAD

**Discussion by twMat** [TiddlyWiki/TiddlyWiki5#6333] [Ideas]
Is the drawing functionality a feature or are we not proud of it? Sure it is a bit rough, but IMO it deserves more prominent recognition and I say we should have a button in the editor toolbar to toggle it. It is poor UX to demand that the user sets the *tiddler type* to toggle the canvas.

Side note: Scribbling in TW is underexploited considering that TW is typically seen as a note taking tool. It would be wonderful to easily make small sketches to accompany ones thoughts during note taking. The current tool is probably not the right one for this but it is a good step in the right direction and if it was more prominent then maybe it would trigger more ideas and development for it.

**Comment by Jermolene** [comment]
Hi @twMat I agree. I've encountered users who had no idea the feature exists.

At some point we did add the page control button that directly creates an image tiddler but one has to hunt for it.

We could offer an alternative to the "new tiddler" button that shows a dropdown with the different types of tiddler that can be created, with different icons for each (we have those icons already).

A similar idea that I've wondered about in the past is to move the "type" dropdown up to the top of the edit template, and make it be a neat row of icons for wikitext, image, markdown etc, with a dropdown for the obscure types.

One trouble is that in general switching the type of an existing tiddler will wipe out the current value of the text field, so we'd want to discourage that, perhaps with a confirmation message.

**Reply by twMat** [reply]
You've clearly pondered over this - I like all of your suggested ideas!

For repositioning+splitting the "type" dropdown, there is a surprisingly obvious empty area in the tiddler toolbar. I'm guessing this area is what you also have in mind. It would (finally) make the edit view much more symmetrical:

<img width="601" alt="image" src="https://user-images.githubusercontent.com/4534056/145892189-b2721af1-6835-480e-aeff-30cb3f1d7d1c.png">

And as you note, they could be pretty *icons* buttons instead of the current type input field+dropdown. If input fields are needed then these could perhaps be the first element inside the dropdown. 

>One trouble is that in general switching the type of an existing tiddler will wipe out the current value of the text field, so we'd want to discourage that, perhaps with a confirmation message.

Again, yes, that makes sense in the same way that you get a confirmation message when attempting to close the edit view by clicking X.
